packages/my-react-package/src/MetaWallet.tsx

```diff
@@ -76,15 +76,14 @@ class MetaWallet {
     /**
              * sign the transaction
              * 
-             * @param {string} scwAddress - Address of the smart contract wallet 
              * @param {string} chainId - chain Id of the transaction
              * @param {any} safeTxBody - the transaction built by TXBuilder 
              * @returns {Promise<string>} the signed transaction
              */
-    async getSignedTX(scwAddress: string, chainId: string, safeTxBody: BuildExecTransaction):Promise<string>{
+    async getSignedTX(chainId: string, safeTxBody: BuildExecTransaction):Promise<string>{
 
         const signature = await this.webWallet._signTypedData({
-            verifyingContract: scwAddress,
+            verifyingContract: this.address,
```

Comment:
this.address should be the address of the MetaWallet. 
The MetaWallet and the SCW have different addresses. You can add another attribute called this.scwAddress